projects/components/src/breadcrumbs/breadcrumbs.service.ts

```diff
@@ -6,8 +6,8 @@ import { concat, defer, EMPTY, Observable, of } from 'rxjs';
 import { catchError, map, shareReplay, switchMap, toArray } from 'rxjs/operators';
 
 @Injectable({ providedIn: 'root' })
-export class BreadcrumbsService {
-  public readonly breadcrumbs$: ReplayObservable<Breadcrumb[]>;
+export class BreadcrumbsService<T extends Breadcrumb> {
```

Comment:
may be easier to default T so references to breadcrumbs service don't change:
```suggestion
export class BreadcrumbsService<T extends Breadcrumb = Breadcrumb> {
```

Reply:
Alternatively, could give breadcrumb an unknown payload and add the defaulted type param to breadcrumb so we can leave the whole usage exactly as is - in that scenario, the only places that would know about any custom data would be the producer and consumer of the breadcrumbs.

Reply:
I feel defaulting to Breadcrumb is cleaner.